modules/internal/ChapelBase.chpl

```diff
@@ -1488,6 +1488,82 @@ module ChapelBase {
       throw new owned TaskErrors(e.errors);
   }
 
+  // Routines for re-interpreting reals as uints and vice-versa, at
+  // the bit level
+  //
+  @unstable("This routine may change names / signatures")
+  proc param (real(64)).transmute(type t) param : t {
+    if t != uint {
+      compilerError("Cannot (currently) transmute from real(64) to " +
+                    t:string);
+    } else {
+      param ui: uint(64) = __primitive("real64 as uint64", this);
+      return ui;
+    }
+  }
+
+  @unstable("This routine may change names / signatures")
+  proc param (real(32)).transmute(type t) param : t {
+    if t != uint(32) {
+      compilerError("Cannot (currently) transmute from real(32) to " +
+                    t:string);
+    } else {
+      param ui: uint(32) = __primitive("real32 as uint32", this);
+      return ui;
+    }
+  }
+
+  @unstable("This routine may change names / signatures")
+  inline proc (real(?w)).transmute(type t): t {
+    use CTypes;
+
+    if t != uint(w) {
+      compilerError("Cannot (currently) transmute from real(" + w:string +
+                    ") to " + t:string);
+    } else {
+      var src = this, dst:uint(w);
+      c_memcpy(c_ptrTo(dst), c_ptrTo(src), w/8);
```

Comment:
Mild preference for `c_sizeof(t)` instead of `w/8`. Or if you want it at chapel compilation time, `numBytes(t)`